allennlp/data/fields/sequence_label_field.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, List, Optional, Union  # pylint: disable=unused-import
+import logging
+
+from overrides import overrides
+import numpy
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.fields.field import Field
+from allennlp.data.fields.sequence_field import SequenceField
+from allennlp.data.vocabulary import Vocabulary
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class SequenceLabelField(Field[numpy.ndarray]):
+    """
+    A ``SequenceLabelField`` assigns a categorical label to each element in a :class:`SequenceField`.
+    Because it's a labeling of some other field, we take that field as input here, and we use it to
+    determine our padding and other things.
+
+    This field will get converted into a list of integer class ids, representing the correct class
+    for each element in the sequence.
+
+    Parameters
+    ----------
+    labels : ``List[str]``
```

Comment:
Update type here, and note that if the input is all ints, we won't use a vocab to convert it.